bindings/java/examples/basic-app/src/main/java/org/example/Migration.java

```diff
@@ -0,0 +1,330 @@
+                    try {
+                        // 0 for default mwm
+                        manager.sendMigrationBundle(nodes, bundleHash, (short) 0);
+                    } catch (Exception e) { 
+                        e.printStackTrace(); 
+                    }
+                }
+                started = true;
+                return mnemonic;
+            } else {
+                System.out.println("Detected 0 balance. Exiting.");
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        this.account = null;
+        return null;
+    }
+
+    private List<List<InputData>> getMigrationBundles(InputData[] inputs){
+        List<InputData> spent = new LinkedList<>();
+        List<InputData> unspent = new LinkedList<>();
+
+        for (InputData input : inputs){
+            if (input.spent()) {
+                spent.add(input);
+              } else {
+                unspent.add(input);
+              }
+        }
+
+        List<List<InputData>> unspentInputChunks = selectInputsForUnspentAddresses(unspent);
+        List<InputData> spentInputs = spent.stream()
+            .filter(input -> input.balance() >= MINIMUM_MIGRATION_BALANCE)
+            .collect(Collectors.toList());
+
+        List<List<InputData>> totalList = new LinkedList<>(); 
+        spentInputs.stream().forEach(i -> totalList.add( Arrays.asList(i) ) );
+        unspentInputChunks.stream().forEach(iList -> totalList.add( iList ) );
+
+        return totalList;
+    }
+
+    /**
+     * Prepares inputs (as bundles) for unspent addresses.
+     * Steps:
+     *   - Categorises inputs in two groups 1) inputs with balance >= MINIMUM_MIGRATION_BALANCE 2) inputs with balance < MINIMUM_MIGRATION_BALANCE
+     *   - Creates chunks of category 1 input addresses such that length of each chunk should not exceed MAX_INPUTS_PER_BUNDLE
+     *   - For category 2: 
+     *         - Sort the inputs in descending order based on balance;
+     *         - Pick first N inputs (where N = MAX_INPUTS_PER_BUNDLE) and see if their accumulative balance >= MINIMUM_MIGRATION_BALANCE
+     *         - If yes, then repeat the process for next N inputs. Otherwise, iterate on the remaining inputs and add it to a chunk that has space for more inputs
+     *         - If there's no chunk with space left, then ignore these funds. NOTE THAT THESE FUNDS WILL ESSENTIALLY BE LOST!
+     * 
+     * NOTE: If the total sum of provided inputs are less than MINIMUM_MIGRATION_BALANCE, then this method will just return and empty array as those funds can't be migrated.
+     * 
+     * This method gives precedence to max inputs over funds. It ensures, a maximum a bundle could have is 30 inputs and their accumulative balance >= MINIMUM_MIGRATION_BALANCE
+     * 
+     * @method selectInputsForUnspentAddresses
+     * 
+     * @params {List<InputData>} inputs
+     * 
+     * @returns {List<List<InputData>>}
```

Comment:
There are syntax highlighting issues; should they be `@param` and `@return` instead?

```suggestion
     * @param {List<InputData>} inputs
     * 
     * @return {List<List<InputData>>}
```